src/scimlfunctions.jl

```diff
@@ -2477,8 +2477,9 @@ function ODEFunction{iip, specialize}(f;
 
     _f = prepare_function(f)
 
-    sys = something(sys, SymbolCache(syms, paramsyms, indepsym))
-
+    if sys === nothing && (syms !== nothing  || paramsyms !== nothing || indepsym !== nothing)
```

Comment:
can these 2 lines be factored out somehow? it feels kind of weird to need to write them so many times.